src/puya/awst/nodes.py

```diff
@@ -209,6 +241,20 @@ def accept(self, visitor: StatementVisitor[T]) -> T:
 
 @attrs.frozen
 class Switch(Statement):
+    """
+    Evaluates `value` and the `key` expression of all cases.
```

Comment:
Whilst this is an important detail, the docstring should start with a high level summary of the functionality.

Suggestion:
> Compare an expression against a set of possible values, and if a match is found, execute the corresponding block. Otherwise, execute the default block (if supplied). Control flow resumes at the next statement, unless interrupted e.g. with a `Goto` node.